File: Backend/Models/Entities/Procurement_Details.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProcionAPI.Models.Entities
{
    public class Procurement_Details
    {
        [Key]
        public int Procurement_Details_ID { get; set; }
        [Required]
        public int EmployeeID { get; set; }
        [Required]
        public int Procurement_Request_ID { get; set; }
        [Required]
        public int Sign_Off_Status_ID { get; set; }
        [Required]
        public int Procurement_Payment_Status_ID { get; set; }
        [Required]
        public string Account_Code { get; set; }
        [Required]
        public int Procurement_Status_ID { get; set; }
        [Required]
        public int Payment_Method_ID { get; set; }
        [ForeignKey("EmployeeID")]
        public Employee Employee { get; set; }
        [ForeignKey("Procurement_Request_ID")]
        public Procurement_Request Procurement_Request { get; set; }
        [ForeignKey("Sign_Off_Status_ID")]
        public Sign_Off_Status Sign_Off_Status { get; set; }
        [ForeignKey("Procurement_Payment_Status_ID")]
        public Procurement_Payment_Status Procurement_Payment_Status { get; set; }
        [ForeignKey("Account_Code, Budget_ID, Category_ID")]
        public Budget_Line Budget_Line { get; set; }
        [ForeignKey("Procurement_Status_ID")]
        public Procurement_Status Procurement_Status { get; set; }
        [ForeignKey("Payment_Method_ID")]
        public Payment_Method Payment_Method { get; set; }

        [MaxLength(32)]
        public string Item_Type { get; set; }

        [MaxLength(32)]
        public string Buyer_Name { get; set; }

        [MaxLength(32)]
        public string Buyer_Email { get; set; }

        public bool Deposit_Required { get; set; }
        public DateTime Full_Payment_Due_Date { get; set; }
        public decimal Total_Amount { get; set; }
        public bool Payment_Made { get; set; }

        [MaxLength(50)]
        public string Comment { get; set; }

        public bool Proof_Of_Payment_Required { get; set; }
    }
}

File: Frontend/src/app/Shared/ProcurementDetails.ts
import { Employee } from "./Employee";
import { Procurement_Request } from "./Procurement_Request";
import { Sign_Off_Status } from "./SignOffStatus";
import { Procurement_Payment_Status } from "./ProcurementPaymentStatus";
import { BudgetLine } from "./BudgetLine";
import { Procurement_Status } from "./ProcurementStatus";
import { Payment_Method } from "./PaymentMethod";
import { Branch } from "./Branch";

export interface Procurement_Details {
     procurement_Details_ID: number;
     employeeID: Number;
     procurement_Request_ID: number;
     sign_Off_Status_ID: number;
     procurement_Payment_Status_ID: number;
     BudgetLineId: number;
     procurement_Status_ID: number;
     payment_Method_ID: number;
     branch_ID: number;
     branch:Branch;
     employee: Employee;
     procurement_Request: Procurement_Request;
     sign_Off_Status: Sign_Off_Status;
     procurement_Payment_Status: Procurement_Payment_Status;
     budget_Line: BudgetLine;
     procurement_Status: Procurement_Status;
     payment_Method: Payment_Method;
     item_Type: string;
     buyer_Name: string;
     buyer_Email: string;
     deposit_Required: boolean;
     full_Payment_Due_Date: Date;
     total_Amount: Number;
     payment_Made: boolean;
     comment: string;
     proof_Of_Payment_Required: boolean;
     itemReceived: boolean;

}
